merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/LocalMissionModelService.java

```diff
@@ -304,7 +309,24 @@ public void refreshResourceTypes(final String missionModelId)
   throws NoSuchMissionModelException {
     try {
       final var model = this.loadAndInstantiateMissionModel(missionModelId);
-      this.missionModelRepository.updateResourceTypes(missionModelId, model.getResources());
+      final var unitMap = new HashMap<String, String>();
+
+      // Resource type units can be regex so go through our resources and apply the matching units.
+      for (final var resource : model.getResources().keySet()) {
+        for (final var unitString : model.getResourceTypeUnits().keySet()) {
+          // Check validity of the regex.
+          try {
+            Pattern.compile(unitString);
```

Comment:
I think of regex compilation as a somewhat expensive operation - we may want to consider swapping the inner and outer loop